.github/workflows/reusable_mypy_primer_comment.yml

```diff
@@ -1,11 +1,11 @@
 name: Comment with mypy_primer diff
 
 on:
-  workflow_run:
-    workflows:
-      - Run mypy_primer
-    types:
-      - completed
+  workflow_call:
+    inputs:
+      workflow_id:
+        required: true
+        type: string
```

Comment:
Will this workflow continue to work for pull requests from forks?

As I understand it, `workflow_call` is in the forked repo scope, thus I think `pull-requests: write` won't have access to the PR to leave a comment. Believe that's the reason `workflow_run` was used here in the first place as that's executed in the upstream repo scope.

_Note: This PR is from an upstream branch, so it doesn't test what happens to PRs from forks._

Reply:
I've created https://github.com/python/mypy/pull/18414 to test how forks work.

Reply:
Looks like that this is true:

```
Diff from mypy_primer:

RequestError [HttpError]: Resource not accessible by integration
    at /home/runner/work/_actions/actions/github-script/v7/dist/index.js:9537:21
Error: Unhandled error: HttpError: Resource not accessible by integration
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async eval (eval at callAsyncFunction (/home/runner/work/_actions/actions/github-script/v7/dist/index.js:35424:16), <anonymous>:36:1)
    at async main (/home/runner/work/_actions/actions/github-script/v7/dist/index.js:35522:20) {
  status: 403,
```